Cython/Compiler/ExprNodes.py

```diff
@@ -4749,9 +4749,29 @@ def analyse_types(self, env, getting=True):
                 analyse_base=False)
 
         if self.start:
-            self.start = self.start.analyse_types(env)
+            self.start = CondExprNode(
+                self.start.pos,
+                true_val = IntNode(self.start.pos, value = '0'),
```

Comment:
You can add `constant_result=0` here to give it a constant compile time result.

Reply:
Added, and I also added `sys.maxsize` as a `constant_result` for the stop index. That value appears to have been set equivalent to `PY_SSIZE_T_MAX` in all versions of CPython 2.7 through 3.7; and is backed by the official Python documentation.